Release

thoom76 opened the pull request.

Commit:
Added feature-lidar for the lidar module.

Commit:
Set-up the library for uart & autocomplete + started writing some beginning code

Commit:
Now receiving lidar UART with good packages! + Added comments in code

Commit:
Created working protocol interpreter for the lidar data over uart

Commit:
Fixed checksum check to work propperly

Commit:
Converted code in main to code in classes + doxygen comments

Commit:
Created code that saves 360 degree measurements in an array and prints it on the terminal + added more doxygen comments for added code

Lennart99 committed:
move classes to own cpp and hpp file, fix formatting of the files(clang-format doesn't seem to work correctly)

Commit:
Seperated the packet structs from the lidar.hpp to lidar_packets.hpp

Lennart99 committed:
make lidar_c use the usart interface instead of hardware-usart, fix compile errors with the test folder

Lennart99 committed:
Merge branch 'feature-lidar' of https://github.com/R2D2-2019/distance_sensor into feature-lidar

Lennart99 committed:
start on unit-tests for the lidar + some small fixes(mainly formating)

Lennart99 committed:
improve final test, improve documentation of mock_usart

Commit:
Changed README.md to be correct

Lennart99 committed:
change namespace, add constants for message validation, change if statement to return statement

Commit:
Changed little code conform feedback of the pull request

Commit:
Quick fix of renamed variables (it is late already)

Commit:
Merge pull request #1 from R2D2-2019/feature-lidar

Feature lidar

gerritvanos commented on code/headers/lidar.hpp:
```diff
@@ -0,0 +1,118 @@
+#include <usart_connection.hpp>
+
+namespace r2d2 {
+    namespace distance {
```
maybe consider namespace chaining this is a bit less typing and error prone, `namespace r2d2::distance { `is sufficient

gerritvanos commented on code/headers/lidar_packets.hpp:
```diff
@@ -0,0 +1,45 @@
+
+namespace r2d2 {
+    namespace distance {
+        /**
```
same as other file

gerritvanos commented on test/headers/mock_usart.hpp:
```diff
@@ -0,0 +1,54 @@
+#pragma once
```
i believe there is a mock usart in the library at the moment

Lennart99 replied:
There is, but that implementation doesn't have the needed behaviour.
This implementation had the receive() method just always return 0xAA.

https://github.com/R2D2-2019/USART_library/issues/10

gerritvanos replied:
this will be changed shortly we will pick this up with the new team

PatrickDekker98 commented on test/headers/mock_usart.hpp:
```diff
@@ -0,0 +1,54 @@
+
+namespace r2d2 {
+
+    class mock_usart_c : public usart_connection_c {
```
Whats a mock usart doing in your feature, it is terribly misplaced and should probaply be moved to the usart library

gerritvanos replied:
Since this module has be transferd i think it is best the new team(which im a part off) change the implementation to the new library implementation, since it works at this point in time we might need to merge it as is and change it as soon as possible when this is implemented in the library

nicovanbentum commented on code/headers/lidar.hpp:
```diff
@@ -0,0 +1,118 @@
+        public:
+            lidar_packet_header_s header; // This is the fixed amount of data
+                                          // every packet starts with.
+            measurement_data_s measurements[720]; // 720 = 360 * 2. So we sample
```
consider templating for flexibility

nicovanbentum commented on test/main.cpp:
```diff
@@ -0,0 +1,134 @@
+#include "ostream"
```
should be <>

Joepieler committed:
Replace the uart library with the hardware uart.

JoostvanBussum committed:
added code to be able to send a measuered distance(right now the smallest value besides 0) over the canbus

JoostvanBussum committed:
updated code to new match library refactoring

JoostvanBussum committed:
fixed code for the demo, isn't fully working because of the usart library we only get 1 byte of data

JoostvanBussum committed:
added some comments. Code is still bugged(see top lines of Lidar.cpp)

JoostvanBussum committed:
updated to newest usart library, code is bugged because it doesn't seem to enter the main function

TGAssen committed:
Added << operators for the lidar structs

JoostvanBussum committed:
code goes in the main again and added/changed some comments

ThePatman3 committed:
Updated to the new usart library and fixed a linker error

ThePatman3 committed:
Merge branch 'feature-canbuspacket' into feature-interpret_data

Official12345 committed:
Merge pull request #3 from R2D2-2019/feature-interpret_data

Feature interpret data

ThePatman3 committed:
Updated the tests to the new usart

ThePatman3 committed:
Created the tests for the module_c class, though its still bugged

The error message is: The procedure entry point _ZSt19uncaught_exceptionsv could not be located in the dynamic link library. I have been trying to solve it for some time now with little results. The probleem seems to be the accessing the frame_data.mm variable.

ThePatman3 committed:
Removed the bugged module_c test case

ThePatman3 committed:
Added doxygen documentation

ThePatman3 committed:
Applied clang-format on all header and source files

Gian1080 committed:
creating new branch commit

Gian1080 committed:
deleted redundant files

Gian1080 committed:
Working distance sensor with HW-lib

Gian1080 committed:
changed the format, added comments, changed namespace

Gian1080 committed:
name change caused file duplication

Gian1080 committed:
duplicate file, caused by branching from release

Gian1080 committed:
Delete HC_SR04.hpp

Gian1080 committed:
updating push, more like a test

Gian1080 committed:
Merge branch 'feature-HC-SR04' of https://github.com/r2d2-2019/distance_sensor into feature-HC-SR04

ThePatman3 committed:
made the lidar variable in module_c a reference

Gian1080 committed:
wrong folder

Gian1080 committed:
making sure correct code is on Git

Gian1080 committed:
Merge branch 'feature-HC-SR04' of https://github.com/r2d2-2019/distance_sensor into feature-HC-SR04

Gian1080 committed:
typo caused error

Gian1080 committed:
removed ::target for travis

Gian1080 committed:
removed ::target for travis

Gian1080 committed:
travis.yml changes

Gian1080 committed:
adding old lidar files, deleted before, but my mistake. For merge request fixing

Gian1080 committed:
added usart:: to remove error in merge

Gian1080 committed:
Merge pull request #4 from R2D2-2019/feature-HC-SR04

Feature hc sr04

ThePatman3 committed:
Removed unused test_module class

ThePatman3 committed:
Merge branch 'release' into feature-canbuspacket

Gian1080 committed:
Merge pull request #5 from R2D2-2019/feature-canbuspacket

Feature canbuspacket is ready to be merged with release.

Gian1080 committed:
started implmenting the UML diagram with @patman

Gian1080 committed:
added changes to the old HC-SR04 class to implement the interface

Gian1080 committed:
working on frame implementation with @ThePatman3

ThePatman3 committed:
Numerous small fixes

Fixes like added missing ; and changing some int16_t to uint16_t.

ThePatman3 committed:
Added the IDistance_sensor_c interface and renamed single_direction_struct_builder to single_direction_struct_builder_c

ThePatman3 committed:
Placed the IDistance_sensor_c and IUnidirectional_distance_sensor_c in the r2d2::distance namespace.

Gian1080 committed:
small tweak to formula

ThePatman3 committed:
Added the lidar_distance_sensor_c class

ThePatman3 committed:
Merge branch 'feature-distance_sensor_interface' of https://github.com/R2D2-2019/distance_sensor into feature-distance_sensor_interface

Gian1080 committed:
bug fixxing

Gian1080 committed:
Merge branch 'feature-distance_sensor_interface' of https://github.com/R2D2-2019/distance_sensor into feature-distance_sensor_interface

Gian1080 committed:
starting sonar array class

Gian1080 committed:
working on sonar array

ThePatman3 committed:
Added doxygen documentation

Doxygen was added for the module_c and IDistance_sensor_c classes.

ThePatman3 committed:
Corrected the doxygen documentation

Changed \detail to the correct \details.

Gian1080 committed:
more sonar array work

Gian1080 committed:
Merge branch 'feature-distance_sensor_interface' of https://github.com/R2D2-2019/distance_sensor into feature-distance_sensor_interface

Gian1080 committed:
small bugg fixx

ThePatman3 committed:
Added doxygen documentation to the lidar_distance_sensor_c class.

Gian1080 committed:
finishes sonar_array

Gian1080 committed:
Merge branch 'feature-distance_sensor_interface' of https://github.com/R2D2-2019/distance_sensor into feature-distance_sensor_interface

Gian1080 committed:
pragma once

Gian1080 committed:
added documentation

Gian1080 committed:
added some doxygen

Gian1080 committed:
Merge pull request #6 from R2D2-2019/feature-distance_sensor_interface

Feature distance sensor interface
